x-pack/plugins/infra/server/routes/metrics_explorer/lib/query_total_grouping.test.ts

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { queryTotalGroupings } from './query_total_groupings';
+
+describe('queryTotalGroupings', () => {
+  const ESSearchClientMock = jest.fn().mockReturnValue({});
+  const defaultOptions = {
+    timerange: {
+      from: 1615972672011,
+      interval: '>=10s',
+      to: 1615976272012,
+      field: '@timestamp',
+    },
+    indexPattern: 'testIndexPattern',
+    metrics: [],
+    dropLastBucket: true,
+    groupBy: ['testField'],
+  };
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should return 0 when there is no groupBy', async () => {
+    const { groupBy, ...options } = defaultOptions;
```

Comment:
What?!?! You can do that? I've never seen that done and it's a very cool way to do an omit.